### ISSUE TITLE
Plan for the future

### ISSUE THREAD

**Issue by Furnyr** [Furnyr/Dissonity#6]
## TL;DR

This implementation will remain as is (npm package + Unity package) at least during the developer preview. Once the SDK is fully released, I plan to release a new version that won't specifically need a Node.js backend (only the Unity package).

## Plan

I've been researching for a few days and discovered another possible implementation that wouldn't require an npm package. Instead of instantiating the SDK inside Node.js and sharing data with Unity through messages, the Dissonity package could act as the SDK instance. You would retrieve an access token from your backend and use it to authenticate the user inside Unity. The advantages are many:

- Any backend could be used, not only Node.js
- There would be no need to maintain another package (npm)
- The process to receive data would be more simple
- It would be way easier to set up (e.g. the "unityInstance" part wouldn't be necessary)

Although there are also possible disadvantages, since it could be harder to implement new functionality. The npm package would still be maintained for some time to give users time to upgrade.

##

If you have any idea or comments, feel free to contribute in this issue!

Shout-out to [kuylar](https://github.com/kuylar) for her Godot implementation and xelnath on Discord that made me rethink the current implementation lol.

**Comment by Furnyr** [comment]
Obviously this would be a breaking change, so I plan to continue supporting 1.x.x even after 2.0 comes out. Version 2.0 won't need the npm package and will be inside a different folder like `v2` while the version 1.x.x will continue to be inside the `unity` folder.

**Comment by Furnyr** [comment]
Related to the new CSP update, I'm aware that Dissonity v1 only wraps around the `DiscordSDK` class, so it loses some functionality like the `patchUrlMappings` function.

These features will be implemented in Dissonity v2 and might be later available for v1.

**Comment by Furnyr** [comment]
- [Activities General Availability
](https://discord.com/developers/docs/change-log#activities-general-availability)
- Furnyr/Dissonity#11

;>
